nilearn/datasets/tests/test_utils.py

```diff
@@ -275,37 +274,46 @@ def test_filter_columns():
 
 
 def test_uncompress():
-    # Create dummy file
-    fd, temp = mkstemp()
-    os.close(fd)
-
     # Create a zipfile
     dtemp = mkdtemp()
     ztemp = os.path.join(dtemp, 'test.zip')
     ftemp = 'test'
-    with contextlib.closing(zipfile.ZipFile(ztemp, 'w')) as testzip:
-        testzip.writestr(ftemp, 'test')
-    datasets.utils._uncompress_file(ztemp, verbose=0)
-    assert(os.path.exists(os.path.join(dtemp, temp)))
-    shutil.rmtree(dtemp)
-
-    dtemp = mkdtemp()
-    ztemp = os.path.join(dtemp, 'test.tar')
-    with contextlib.closing(tarfile.open(ztemp, 'w')) as tar:
-        tar.add(temp)
-    datasets.utils._uncompress_file(ztemp, verbose=0)
-    assert(os.path.exists(os.path.join(dtemp, temp)))
-    shutil.rmtree(dtemp)
-
-    dtemp = mkdtemp()
-    ztemp = os.path.join(dtemp, 'test.gz')
-    f = gzip.open(ztemp, 'wb')
-    f.close()
-    datasets.utils._uncompress_file(ztemp, verbose=0)
-    assert(os.path.exists(os.path.join(dtemp, temp)))
-    shutil.rmtree(dtemp)
-
-    os.remove(temp)
+    try:
+        with contextlib.closing(zipfile.ZipFile(ztemp, 'w')) as testzip:
+            testzip.writestr(ftemp, 'test')
+        datasets.utils._uncompress_file(ztemp, verbose=0)
+        assert (os.path.exists(os.path.join(dtemp, ftemp)))
+        shutil.rmtree(dtemp)
+
+        dtemp = mkdtemp()
+        ztemp = os.path.join(dtemp, 'test.tar')
+        ftemp = 'test'
+
+        # Create dummy file in the dtemp folder, so that the finally statement
+        # can easily remove it
+        fd, temp = mkstemp(dir=dtemp)
+        os.close(fd)
+        with contextlib.closing(tarfile.open(ztemp, 'w')) as tar:
+            tar.add(temp, arcname=ftemp)
+        datasets.utils._uncompress_file(ztemp, verbose=0)
+        assert (os.path.exists(os.path.join(dtemp, ftemp)))
+        shutil.rmtree(dtemp)
+
+        dtemp = mkdtemp()
+        ttemp = os.path.join(dtemp, 'test')
+        with open(ttemp, 'wb') as f:
+            f.write('test'.encode('utf-8'))
```

Comment:
I could not understand the point of write extra ```ttemp``` ?

If I understood correctly, you are testing the uncompression of test.gz and after uncompressing you are missing to test ftemp as what you have done above.

Reply:
you are right. `ttemp` would exist anyway. 
to get a bytes buffer without creating a file you can use `io.BytesIO`. you could also just use the `GzipFile` (i.e. `f_out`)'s `write` method. but actually it is not necessary to write anything if you're not checking the contents of the uncompressed file afterward (it will exist even if the compressed file was empty)

Reply:
yes you're right. I'll create an empty BytesIO then. not sure we need to check the content, since the pure uncompressing bit is done by python libraries, although I can do it if needed.

Reply:
you don't need to 
```
with gzip.open(ztemp, 'wb') as f_out:
    pass
```
would be fine if you don't check the content,
```
with gzip.open(ztemp, 'wb') as f_out:
    f_out.write(b"this should be found")
```
otherwise

but I think the test is fine as it is, now it will pass if and only if test.gz gets uncompressed. thanks!